### PR TITLE
Handle remaining todos in code

### PR DESCRIPTION
**Pull request by matthewloring**

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots PTAL.

File: lib/trace-agent.js [hunk 1]
```diff
@@ -234,7 +236,11 @@ TraceAgent.prototype.addContextToHeaders = function(spanData, headers) {
 module.exports = {
   get: function(config, logger) {
     if (traceAgent) {
-      // TODO: log error if config object is different from traceAgent.config
+      if (!_.isEqual(config, traceAgent.config_)) {
+        traceAgent.logger.warn('New configuration does not match configuration' +
+          'of existing agent:\nNew: ' + JSON.stringify(config) + '\nExisting: ' +
+          JSON.stringify(traceAgent.config_));
```
Comment (ofrobots) [inline lib/trace-agent.js]: The message needs to inform the user that their old config is still in effect.

File: package.json [hunk 2]
```diff
@@ -50,6 +50,7 @@
   "dependencies": {
     "@google/cloud-diagnostics-common": "^0.2.1",
     "continuation-local-storage": "^3.1.4",
+    "lodash": "^4.0.0",
```
Comment (ofrobots) [inline package.json]: Do you want to use `lodash.isequal` instead?

**Comment by ofrobots** [comment]
lgtm w/ comments.

File: lib/trace-agent.js (hunk 1 above)
Comment (matthewloring) [inline lib/trace-agent.js]: Done.

File: package.json (hunk 2 above)
Comment (matthewloring) [inline package.json]: Didn't realize this was out there. Definitely a better fit.

**Pull request merged**